xdsl/transforms/experimental/stencil_global_to_local.py

```diff
@@ -0,0 +1,72 @@
+from dataclasses import dataclass
+from typing import TypeVar
+from abc import ABC, abstractmethod
+
+from xdsl.pattern_rewriter import (PatternRewriter, PatternRewriteWalker,
+                                   RewritePattern, GreedyRewritePatternApplier,
+                                   op_type_rewrite_pattern)
+from xdsl.ir import MLContext
+from xdsl.irdl import Attribute
+from xdsl.dialects import builtin
+from xdsl.dialects.experimental import stencil
+
+_T = TypeVar('_T', bound=Attribute)
+
+
+@dataclass
+class SlicingStrategy(ABC):
+    @abstractmethod
+    def calc_resize(self, shape: tuple[int]) -> tuple[int]:
+        raise NotImplementedError("SlicingStrategy must implement calc_resize!")
+
+
+@dataclass
+class HorizontalSlices2D(SlicingStrategy):
+    slices: int
+
+    def __post_init__(self):
+        assert self.slices > 1, "must slice into at least two pieces!"
```

Comment:
As a general comment, I guess we need to do this here as the stencil size itself is static - but in general you wouldn't want to specify this. Instead you would want this to be equal to the number of MPI ranks (but the challenge there is that makes your local size dynamic which the stencil dialect currently can't handle I don't think?)